asset-transfer-basic/application-typescript-hsm/README.md

```diff
@@ -0,0 +1,240 @@
+# Asset Transfer Basic typescript HSM sample
+
+This sample takes the Asset Transfer basic example and re-works it to focus on how
+you would use a Hardware Security Modules within your node client application.
+
+## About the Sample
+
+This typescript sample application is able to use any of the asset transfer basic
+chaincode samples. It will show how to register users with a Fabric CA and enroll users
+which will store keys in an HSM (In this case the sample uses SoftHSM which is an HSM implementation that should be used
+for development and testing purposes only). It also demonstrates setting up a wallet to
+interact that will store identities that can then be used to transact on the fabric network which
+are HSM managed.
+## C Compilers
+In order for the client application to run successfully you must ensure you have C compilers and Python3
```

Comment:
I would have to take guidance from someone who has a Mac. I state you need python3 because that's the only supported version. No one should be using python2 and I don't want to advocate it, especially if node-gyp decides to drop support for python2

Reply:
I've added a note to say that Python 2 could still work however it is not a supported environment anymore and may stop working in the future

Reply:
Many thanks for all the useful feedback, really appreciated